packages/features/data-table/lib/resizing.ts

```diff
@@ -49,8 +49,13 @@ function getAdjustedColumnSizing<TData>({
     return currentColumnSizing;
   }
 
-  const getColumnSize = (id: string) =>
-    resizedColumns.has(id) ? currentColumnSizing[id] : initialColumnSizing[id];
+  const getColumnSize = (header: Header<TData, unknown>) => {
```

Comment:
The addition of the `columnVisibility` state revealed a bug that didn't exist without `columnVisibility`. So I made some changes here.